Source/aweXpect.Reflection/ThatMethods.Return.cs

```diff
@@ -0,0 +1,84 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+using System.Text;
+using aweXpect.Core;
+using aweXpect.Core.Constraints;
+using aweXpect.Reflection.Helpers;
+using aweXpect.Results;
+
+// ReSharper disable PossibleMultipleEnumeration
+
+namespace aweXpect.Reflection;
+
+public static partial class ThatMethods
+{
+	/// <summary>
+	///     Verifies that all methods in the filtered collection return type <typeparamref name="TReturn" />.
+	/// </summary>
+	public static AndOrResult<IEnumerable<MethodInfo>, IThat<IEnumerable<MethodInfo>>> Return<TReturn>(
```

Comment:
You should return a dedicated class that allows specifying alternative return types, e.g.
```csharp
.Return<Task>().OrReturn<ValueTask>()
```

For this you need to start the list of possible return types in the first call of "Return" and forward it to the returned classes, so that the new types can be added to the list there.